src/gui/transferlistwidget.cpp

```diff
@@ -269,6 +269,15 @@ void TransferListWidget::torrentDoubleClicked()
         else
             torrent->pause();
         break;
+    case PREVIEW_FILE:
+        if (torrentContainsPreviewableFiles(torrent)) {
+            auto *dialog = new PreviewSelectDialog(this, torrent);
+            dialog->setAttribute(Qt::WA_DeleteOnClose);
+            connect(dialog, &PreviewSelectDialog::readyToPreviewFile, this, &TransferListWidget::previewFile);
+            dialog->show();
+            break;
+        }
+        /* fall-through */
```

Comment:
I think it is better not doing "fall-through" here. I would wrap `OPEN_DEST` case into an anonymous function and use it as needed.